File: JavaCheatSheet/src/main/java/com/github/jameswdelancey/javaCheatSheet/Part4.java
package com.github.jameswdelancey.javaCheatSheet;

public class Part4 {
    public static void main(String[] args) {
        System.out.println("Hello, world!");
        int x = 17;
        int y = 9;
        findAnswer();
        int answer = findAnswer2(x, y);
        System.out.println("The answer is: " + answer);
    }

    public static void findAnswer() {
        int answer = 42;
        System.out.println("The answer is " + answer);
    }

    public static int findAnswer2(int x, int y) {
        return x + y + 5;
    }
}

File: rustCheatSheet/src/part4.rs
#[test]
fn main() {
    println!("Hello, world!");
    let x = 17;
    let y = 9;
    let _ = findAnswer();
    let answer = findAnsweri2(x, y);
    println!("The answer is: {}", answer);
}

fn findAnswer() {
    let answer = 42;
    println!("The answer is {}", answer);
}

fn findAnsweri2(x: i32, y: i32) -> i32 {
    x + y + 5
}

#[cfg(test)]
mod tests {
    use crate::part4::findAnsweri2;

    fn testFindAnswer() {
        let x = 5;
        let y = 7;
        let answer = findAnsweri2(x, y);
        assert_eq!(17, answer);
    }
}
